share/csm_share/shr/shr_flux_mod.F90

```diff
@@ -267,7 +277,15 @@ real(R8)    :: psix2  ! stability function at ztref reference height
      if (mask(n) /= 0) then
     
         !--- compute some needed quantities ---
-        vmag   = max(umin, sqrt( (ubot(n)-us(n))**2 + (vbot(n)-vs(n))**2) )
+
```

Comment:
**This is a critical problem here.** CESM does not use these variables (ugust and gprec) and these changes will change a very fundamental calculation for CESM. These need to be brought into this routine in a way that permits backwards compatibility with the old calculation.